clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiAdapter.groovy

```diff
@@ -138,14 +140,40 @@ class KubernetesClientApiAdapter {
 
   List<V1beta1StatefulSet> getStatefulSets(String namespace) {
     exceptionWrapper("statefulSets.list", "Get Stateful Sets", namespace) {
+      /*
+       "fixme" and note this is k8s-client api issue and we are working this as workaround.
+        */
       V1beta1StatefulSetList list = apiInstance.listNamespacedStatefulSet(namespace, null, null, null, null, API_CALL_TIMEOUT_SECONDS, false)
+      String apiVersion = list.getApiVersion();
+      String kind = list.getKind();
+      if (kind != null && kind.equals("StatefulSetList")) {
```

Comment:
This seems very specific - shouldn't you be annotating the kind of each returned resource with `StatefulSet` no matter what?